src/useKeyboardEvent/useKeyboardEvent.ts

```diff
@@ -0,0 +1,42 @@
+import { DependencyList, RefObject, useMemo } from 'react';
+import { useEventListener } from '..';
+import { noop, isBrowser } from '../util/const';
+import { yieldTrue, yieldFalse } from '../util/misc';
+
+export type IKeyPredicate = (event: KeyboardEvent) => boolean;
+export type IKeyFilter = null | undefined | string | IKeyPredicate;
+export type IHandler = (event: KeyboardEvent) => void;
+
+export type IUseKeyOptions<T extends EventTarget> = {
+  event?: 'keydown' | 'keypress' | 'keyup';
+  target?: RefObject<T> | T | null;
+  eventOptions?: boolean | AddEventListenerOptions;
+};
+
+const createKeyPredicate = (keyFilter: IKeyFilter): IKeyPredicate => {
+  if (typeof keyFilter === 'function') return keyFilter;
+  if (typeof keyFilter === 'string') return (event: KeyboardEvent) => event.key === keyFilter;
+  return keyFilter ? yieldTrue : yieldFalse;
+};
+export function useKeyboardEvent<T extends EventTarget>(
+  keyOrPredicate: IKeyFilter,
+  fn: IHandler = noop,
```

Comment:
Mismatch with docs.
Also make jsdocs comments please,  it is easier to use in IDE, since no need to go to documentation.